metrics/firebase/docs/features/builds_aggregation/01_firestore_builds_aggregation.md

```diff
@@ -1,39 +1,169 @@
+- [Firestore Cloud Function using Dart](https://github.com/Flank/flank-dashboard/blob/concatenate_document_sections/metrics/firebase/docs/features/dart_cloud_functions/01_using_dart_in_the_firebase_cloud_functions.md)
+
+## Contents
+
+- [**Analysis**](#analysis)
+    - [Feasibility study](#feasibility-study)
+    - [Requirements](#requirements)
+    - [Landscape](#landscape)
+      - [Loading all documents](#loading-all-documents)
+      - [Aggregating data using the Cloud Functions](#aggregating-data-using-the-cloud-functions)
+        - [Firebase transactions](#firebase-transactions)
+        - [FieldValue.increment](#fieldvalueincrement)
+        - [Distributed counter](#distributed-counter)
+      - [Decision](#decision)   
+- [**Design**](#design)
+    - [Database](#database)
+      - [build_days](#build_days)
+        - [Document Structure](#document-structure)
+        - [Security Rules](#security-rules)
+      - [tasks](#tasks)
+        - [Document Structure](#document-structure-1)
+        - [Security Rules](#security-rules-1)
+    - [Program](#program)
+      - [onBuildAdded](#onBuildAdded)
+      - [onBuildUpdated](#onBuildUpdated)
+      - [Testing](#Testing)
+
+# Analysis
+> Describe a general analysis approach.
+
+During the analysis stage, we should understand a [feasibility](#feasibility-study) to implement the `builds aggregation` to reduce Firebase usage. For this purpose, we should investigated a possibility to provide some aggregation calculations using the back-end, provided by the Firebase using the Cloud Functions.
+
+Based on the analysis, we should make a decision about an optimal approach that satisfies the [requirements](#requirements) for the Firestore builds aggregation.
+
+### Feasibility study
+> A preliminary study of the feasibility of implementing this feature.
+
+If we want to quickly find documents in large collections, we should use Firebase advanced queries. Advanced queries in Cloud Firestore allow quickly find documents in large collections. If we want to gain insight into the properties of the collection as a whole (e.g., builds per week), we need an aggregation over a collection. Unfortunately, Cloud Firestore does not support native aggregation queries. So, we should implement this calculation because of the Firestore limitations.
+
+For now, to make builds aggregations we should load all builds and process calculations on the client, because Cloud Firestore does not support the aggregation queries.
+
+This solution is okay if a project has not many builds, but over time, the number of builds growth and it will pull along increase the number of reads from the database and leads to heavy load/price implications.
+
+### Requirements
+> Define requirements and make sure that they are complete.
+
+- Reduces Firestore usage.
+- Processes aggregation calculations.
+- Works correctly in case of concurrent document edits.
```

Comment:
:tw: I think it would be great to introduce this list. See [Google Technical Writing](https://developers.google.com/tech-writing/one/lists-and-tables#introduce_each_list_and_table) for more details.